### DISCUSSION TITLE
Shared Dev Services Database?

### DISCUSSION THREAD

**Discussion by nscuro** [quarkusio/quarkus#40351] [Q&A]
Quarkus supports sharing of certain Dev Services containers (AMQP, Kafka) across multiple applications or instances of an application. This is super useful for systems with shared infra!

However, it seems what can not be shared are database containers, which strikes me as a bit odd. Working around this limitation with Docker Compose is possible, but complicates the otherwise so straightforward dev experience.

Is there a particular reason why that is not supported? Is eventually supporting shared DB containers planned? Or has anyone found a way to make it work already?

[a comment by an automated account is not shown]

**Reply by geoand** [reply]
cc @yrodiere

**Comment by yrodiere** [comment]
> Is there a particular reason why that is not supported?

Just that nobody worked on it, I think. From what I understand the feature is strictly about how Quarkus uses Docker, so it doesn't really matter what kind of container is being shared.

>  Is eventually supporting shared DB containers planned? Or has anyone found a way to make it work already?

I didn't see a feature request, feel free to open one. If you want to work on it, the good news is there's pre-existing work to base your work on, and I'll happily review your pull request :)

**Reply by nscuro** [reply]
Thanks @yrodiere, I'll make sure to raise a feature request!

> If you want to work on it [...]

It just so happens that I'll be on PTO starting tomorrow, so I might just take a stab at it :)

**Comment by holly-cummins** [comment]
Note that the [docs](https://quarkus.io/guides/databases-dev-services#reuse-within-execution) are also wrong: 

> Within a dev mode session or test suite execution, Quarkus will always reuse database Dev Services as long as their configuration (username, password, environment, port bindings, …​) did not change.
> 
> When the configuration of any database Dev Services changes, Quarkus will always restart all database Dev Services.
